components/fastly_frontend/main.tf

```diff
@@ -1,7 +1,17 @@
-variable "phoenix_name" {}
-variable "phoenix_backend" {}
-variable "papertrail_destination" {}
-variable "papertrail_log_format" {}
+variable "name" {
+  description = "The name for this Fastly property."
+  type        = string
+}
+
+variable "application" {
+  description = "The application module to route traffic to."
+  type        = object({ name = string, domain = string, backend = string })
+}
```

Comment:
Following the same pattern established in #270, you can just pass in an application module here (as long as it has these named outputs). So we can just say `application = module.phoenix` for example, rather than needing to pass each piece separately! 😌